### ISSUE TITLE
.extend() collision brand (FIX_EXTENSION_COLLISION) + duplicate ExtensionId dedup

### ISSUE THREAD

**Issue by chenxin-yan** [chenxin-yan/crust#280]
## Context

PR #277 removed runtime definition validation in favor of compile-time `FIX_*` brands. `.extend()` is the one registration surface with **no compile-time collision brand**: extension-contributed flag/command/Context names that collide with app-defined names (or with another extension) are silent, deterministic last-write-wins. This was flagged as the residual gap in the PR #277 review and confirmed by the follow-up audit.

```ts
app.flags({ verbose: { type: "boolean", short: "v" } })
   .extend(someExtension()) // also contributes --verbose → silent overwrite
```

App-level `.flags()` / `.add()` collisions already fail typecheck with `FIX_ALIAS_COLLISION` / `FIX_COMMAND_COLLISION`.

## Part 1 — collision brand

1. **Make `defineExtension` carry contributed names as types.** Today `Extension` is a plain structural value — flag/command names are runtime `Record<string, ...>` keys, invisible to the checker. It needs phantom type parameters capturing the literal names, the way `SpellingsOf<F>` already computes flag spellings for builders:
   ```ts
   Extension<ContributedSpellings extends string, ContributedCommands extends string>
   ```
2. **Check at `.extend()`.** The `Crust` builder already accumulates known spellings in `Sp` and sibling command names in `Sibs`. `.extend()` intersects the extension's contributed names against those and brands on overlap: `FIX_EXTENSION_COLLISION`.

Dynamically built extensions (names not statically known) stay last-write-wins — consistent with every other rule.

## Part 2 — duplicate extension id dedup (runtime)

`.extend()` currently accumulates without dedup:

```ts
extensions: [...this._node.extensions, ...extensions],
```

The old "Extension id already registered" runtime `DEFINITION` check was deleted in PR #277. Registering the same extension twice today:

- contributions (commands/flags/contexts) collapse via last-write-wins — harmless
- **hooks run once per registration** — duplicated side effects (update-notifier checks twice, skills auto-update twice)

Since #264 gives every extension a first-class `ExtensionId`, dedup-by-id at `prepareInvocation` (keep the last registration, consistent with last-write-wins) is a small runtime fix. Duplicate ids are type-invisible, so runtime is this rule's one enforcement layer.

## Why deferred

`Crust` already carries 7 generic parameters and the repo has fought type-check-cost regressions before (dedicated `type-performance` CI job; PR #231 trimmed 8 generics → 6). Threading extension name-tracking through `Extension` / `defineExtension` / `.extend()` is the most type-expensive brand of the family — needs benchmark evidence, not just implementation.

## Acceptance

- Statically known extension flag/command collisions with app or prior-extension names fail typecheck with `FIX_EXTENSION_COLLISION`
- Dynamically built extensions keep documented last-write-wins
- Duplicate `ExtensionId` registrations run hooks once (last registration wins); regression test for double-`.extend()` of the same extension
- `bun scripts/type-perf-report.ts` shows acceptable instantiation cost vs main
